src/main.rs

```diff
@@ -4,33 +4,43 @@ extern crate easy_hash;
 
 use easy_hash::{Sha256, Hasher, HashResult};
 use std::thread;
-use std::sync::mpsc;
+use std::sync::{mpsc, Arc};
+use std::sync::atomic::{AtomicBool, Ordering};
 
 const BASE: usize = 42;
 const THREADS: usize = 4;
 static DIFFICULTY: &'static str = "000000";
 
 struct Solution(usize, String);
 
-fn search_for_solution(start_at: usize, sender: mpsc::Sender<Option<Solution>>) {
+fn search_for_solution(start_at: usize, sender: mpsc::Sender<Solution>, is_solution_found: Arc<AtomicBool>) {
+    let mut iteration_no = 0;
     for i in (start_at..).step_by(THREADS) {
         let hash: String = Sha256::hash((i * BASE).to_string().as_bytes()).hex();
-        let result = if hash.ends_with(DIFFICULTY) { Some(Solution(i, hash)) } else { None };
-
-        match sender.send(result) {
-            Ok(_)  => {},
-            Err(_) => {
-                println!("Receiver has stopped listening, dropping worker number {}", start_at);
-                break;
-            },
+        
+        if hash.ends_with(DIFFICULTY) {
+            is_solution_found.store(true, Ordering::Relaxed);
+            match sender.send(Solution(i, hash)) {
+                Ok(_)  => {},
+                Err(_) => println!("Receiver has stopped listening, dropping worker number {}.", start_at),
+            }
+            break;
         }
+         
+        if iteration_no % 1000 == 0 && is_solution_found.load(Ordering::Relaxed) {
```

Comment:
You might as well turn that into an "else if" since you break in the "if" branch anyhow.